### PR TITLE
cleanup: refactor the way extracting Node smaller events

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This PR...
- Moves `NodeSchedulingPropertiesChange` and its dependent functions to event.go
  - Necessary for #122628.
- Simplifies `NodeSchedulingPropertiesChange`.
  - https://github.com/kubernetes/kubernetes/pull/122628/files#r1675521201

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/queue/events.go [hunk 1]
```diff
@@ -92,58 +89,71 @@ var (
...
+	nodeChangeExtracters := []func(newNode *v1.Node, oldNode *v1.Node) *framework.ClusterEvent{
+		extractNodeSpecUnschedulableChange,
+		extractNodeAllocatableChange,
+		extractNodeLabelsChange,
+		extractNodeTaintsChange,
+		extractNodeConditionsChange,
+		extractNodeAnnotationsChange,
 	}
 
+	var events []framework.ClusterEvent
```
Comment (gabesaba) [inline pkg/scheduler/internal/queue/events.go:103]: nit: turn this into a named return value

File: pkg/scheduler/internal/queue/events.go [hunk 2]
```diff
@@ -92,58 +89,71 @@ var (
...
-	}
-	if nodeTaintsChanged(newNode, oldNode) {
-		events = append(events, NodeTaintChange)
-	}
-	if nodeConditionsChanged(newNode, oldNode) {
-		events = append(events, NodeConditionChange)
-	}
-	if nodeAnnotationsChanged(newNode, oldNode) {
-		events = append(events, NodeAnnotationChange)
+	nodeChangeExtracters := []func(newNode *v1.Node, oldNode *v1.Node) *framework.ClusterEvent{
```
Comment (gabesaba) [inline pkg/scheduler/internal/queue/events.go:94]: is there any runtime implication to having this here? if so, does Golang allow it to be defined as a var/const at file level?

File: pkg/scheduler/internal/queue/events_test.go [hunk 3]
```diff
@@ -1,9 +1,12 @@
 /*
 Copyright 2024 The Kubernetes Authors.
+
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
+
     http://www.apache.org/licenses/LICENSE-2.0
+
```
Comment (gabesaba) [inline pkg/scheduler/internal/queue/events_test.go:9]: nit: delete ws

**Review by gabesaba — COMMENTED** [review]
lgtm, modulo a few nits. thanks for making these changes. i think it will make the primary PR much easier to review :)

File: pkg/scheduler/internal/queue/events_test.go (hunk 3 above)
Comment (sanposhiho) [reply pkg/scheduler/internal/queue/events_test.go:9]: All the other files have those empty lines.

**Comment by sanposhiho** [comment]
@gabesaba 
Addressed your points.

**Pull request merged**